ggshield/cmd/honeytoken/create_with_context.py

```diff
@@ -0,0 +1,147 @@
+        from the repository or from OUTPUT_FILE, if set.",
+)
+@click.option(
+    "-o",
+    "--output",
+    "output_file",
+    type=RealPath(file_okay=True, dir_okay=False, readable=True, writable=True),
+    required=False,
+    help="Filename to store your honeytoken.",
+)
+@add_common_options()
+@click.pass_context
+def create_with_context_cmd(
+    ctx: click.Context,
+    name: Optional[str],
+    type_: str,
+    description: Optional[str],
+    output_file: Optional[Path],
+    language: Optional[str],
+    **kwargs: Any,
+) -> int:
+    """
+    Create a honeytoken within a context.
+
+    The context is realistic file in which your honeytoken is inserted.
+    Adding your honeytoken within a relevant context makes it look more credible.
+
+    The prerequisites to use this command are the following:
+
+    - you have the necessary permissions as a user (for now, Honeytoken is
+    restricted to users with a manager role),
+
+    - the personal access token used by ggshield has the `honeytokens:write`
+    scope.
+    """
+    # if name is not given, generate a random one
+    if not name:
+        name = generate_random_honeytoken_name()
+    ctx_obj = ContextObj.get(ctx)
+    client = create_client_from_config(ctx_obj.config, ctx_obj.ui)
+    common_options = {"name": name, "type_": type_, "description": description}
+
+    if language or output_file:
+        options = {
+            **common_options,
+            "language": language,
+            "filename": output_file.name if output_file else None,
+        }
+    else:
+        options = {
+            **common_options,
+            "project_extensions": _get_most_common_extensions_in_dir(Path(".")),
+        }
+
+    response = client.create_honeytoken_with_context(**options)
+    if not isinstance(response, (Detail, HoneytokenWithContextResponse)):
+        raise UnexpectedError("Unexpected honeytoken response")
+
+    if isinstance(response, Detail) and response.status_code == 403:
+        raise UnexpectedError(
+            """ggshield does not have permissions to create honeytokens on your workspace. Make sure that:
+
+- the honeytoken module is enabled for your GitGuardian workspace,
+- you have the necessary permissions as a user,
+- the personal access token used by ggshield has the `honeytokens:write` scope.
+
+To learn more, visit https://docs.gitguardian.com/honeytoken/getting-started."""
+        )
+    elif isinstance(response, Detail):
+        raise UnexpectedError(response.detail)
+
+    filepath_to_write = Path(".").joinpath(response.filename)
+
+    click.echo(
+        "Honeytoken created and places in the following context:\n"
+        f"- file: {filepath_to_write.absolute()}\n"
+        f"- language: {response.language} \n"
+        "\n"
+        f'Your honeytoken "{name}" is accessible in your workspace: {response.gitguardian_url}'
+    )
+
+    with filepath_to_write.open("w") as file_to_write:
+        file_to_write.write(response.content)
```

Comment:
Can be simplified to:

```suggestion
    filepath_to_write.write_text(response.content)
```